dll/cpl/appwiz/createlink.c

```diff
@@ -447,8 +449,33 @@ FinishDlgProc(HWND hwndDlg,
             pContext = (PCREATE_LINK_CONTEXT) GetWindowLongPtr(hwndDlg, DWLP_USER);
             if (lppsn->hdr.code == PSN_SETACTIVE)
             {
-                /* TODO: Use shell32!PathCleanupSpec instead of DoConvertNameForFileSystem */
+                /* Remove invalid characters */
                 DoConvertNameForFileSystem(pContext->szDescription);
+
+                /* Is it empty? (rare case) */
+                if (!pContext->szDescription[0])
+                {
+                    HMODULE hShell32 = GetModuleHandleW(L"shell32.dll");
+#define IDS_NEWITEMFORMAT 150
+#define IDS_LNK_FILE 215
```

Comment:
Just a question, no request to act yet, please hear other devs opinions first: Wouldn't it be better to include shell32/shresdef.h somehow instead of redefining those constants here? Or is such an include a no-go? Or is that tricky with our buildsystem somehow? What if those would ever go offsync? E.g. if we will change the target in a few years...

Reply:
`error: "IDC_SHORTCUT_LOCATION" redefined [-Werror]`

Reply:
Renamed that macro in `appwiz`.